File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        if len(s) == 0 or len(s) == 1:
            return False

        for i in range(len(s)):
            if s[i] == '{' or s[i] == '[' or s[i] == '(':
                stack.append(s[i])
            
            if len(stack) == 0:
                return False
            
            top = stack[-1]
            if s[i] == "}" and top != "{":
                return False
    
            if s[i] == ")" and top != "(":
                return False

            if s[i] == "]" and top != "[":
                return False

            if s[i] == "}" and top == "{":
                stack.pop()
                continue
    
            if s[i] == ")" and top == "(":
                stack.pop()
                continue

            if s[i] == "]" and top == "[":
                stack.pop()
                continue
                
        if len(stack) != 0:
            return False

        return True

        

File: 0020-valid-parentheses/0020-valid-parentheses.java
import java.util.*;

class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        if (s.length() < 2){
            return false;
        }

        for( int i = 0 ; i < s.length() ; i++ ){
            if ( s.charAt(i) == '[' || s.charAt(i) == '(' || s.charAt(i) == '{' ){
                stack.push(s.charAt(i));
            }
            if ( stack.isEmpty()){
                return false;
            }

            if ( s.charAt(i) == ']'){
                if ( stack.pop() != '[' ){
                    return false;
                }
            }
            if ( s.charAt(i) == ')'){
                if ( stack.pop() != '(' ){
                    return false;
                }
            }
            if ( s.charAt(i) == '}'){
                if ( stack.pop() != '{' ){
                    return false;
                }
            }
        }
        return stack.isEmpty();
    }
}